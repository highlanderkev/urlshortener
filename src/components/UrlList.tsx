import React from "react";
import PropTypes from "prop-types";
import UrlListItem from "./UrlListItem";
import { URLEntry } from "../urlEntry.model";

type UrlListProps = {
  list: Array<URLEntry>;
  onRemove: Function;
};

class UrlList extends React.Component<UrlListProps> {
  render() {
    return (
      <div>
        <h2>Previously Shortened URLs</h2>
        <ol className="list-group list-group-numbered">
          {this.props.list.map((el: URLEntry) => {
            return (
              <UrlListItem
                key={el.slug}
                item={el}
                onRemove={this.props.onRemove}
              />
            );
          })}
        </ol>
      </div>
    );
  }
}

export default UrlList;
