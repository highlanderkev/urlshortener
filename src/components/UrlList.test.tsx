import React from 'react';
import UrlList from "./UrlList";
import renderer from "react-test-renderer";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()})
describe('URL List Component', () => {
  it("matches snapshot", () => {
    const urlList = renderer.create(<UrlList list={[{ url: 'test', slug: 'test', short_url: 'test'}]} onRemove={() => {}} />);
    let tree = urlList.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays list of url entries", () => {
    const urlList = shallow(<UrlList list={[{ url: 'test', slug: 'test', short_url: 'test'}]} onRemove={() => {}} />)
    expect(urlList.text()).toEqual("Previously Shortened URLs<UrlListItem />");
  });
})
