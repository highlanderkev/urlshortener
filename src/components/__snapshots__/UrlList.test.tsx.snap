// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URL List Component matches snapshot 1`] = `
<div>
  <h2>
    Previously Shortened URLs
  </h2>
  <ol
    className="list-group list-group-numbered"
  >
    <li
      className="list-group-item d-flex justify-content-between align-items-start"
    >
      <div
        className="ms-2 me-auto"
      >
        <div
          className="fw-bold"
        >
          <a
            href="test"
          >
            test
          </a>
          <br />
          <a
            href="test"
          >
            test
          </a>
        </div>
        <small>
          test
        </small>
      </div>
      <button
        aria-label="Close"
        className="btn-close"
        onClick={[Function]}
        type="button"
      />
    </li>
  </ol>
</div>
`;
